src/site/templates.rs

```diff
@@ -0,0 +1,97 @@
+//! Fundamentals for oranda's templating system.
+//!
+//! At the core, oranda uses a minijinja (Jinja2 template)-based system, where all templates are
+//! loaded from memory at the start of runtime. Templates can then be referenced from anywhere within
+//! the application, provided the `Templates` struct is properly passed around. Templates themselves
+//! can also use features such as imports, inheritance, extends, and so on.
+
+use crate::config::Config;
+use crate::data::artifacts::inference::triple_to_display_name;
+use crate::errors::Result;
+use crate::site::layout::LayoutContext;
+use crate::site::markdown::SyntaxTheme;
+use crate::site::{link, markdown};
+use include_dir::{include_dir, Dir};
+use minijinja::value::Value;
+use minijinja::{context, AutoEscape, Environment, Template};
+use std::collections::HashMap;
+
+const TEMPLATE_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/templates");
+
+/// Main templates struct that gets passed around in the application.
+pub struct Templates<'a> {
+    /// Minijinja environment that contains all loaded templates
+    pub env: Environment<'a>,
+    /// Layout context used for every render
+    pub layout: LayoutContext,
+}
+
+impl<'a> Templates<'a> {
+    pub fn new(config: &Config) -> Result<Self> {
+        let mut env = Environment::new();
+        let mut files = HashMap::new();
+        Self::load_files(&TEMPLATE_DIR, &mut files).unwrap();
+        for (path, contents) in files {
+            env.add_template_owned(path, contents).unwrap();
+        }
```

Comment:
these shouldn't ever fail in prod but it would be nice to make these expects to save debugging time when developing?